src/commonComponents/cart/Cart.js

```diff
@@ -0,0 +1,75 @@
+// prop-types
+import PropTypes from "prop-types";
+
+// lodash
+import _noop from "lodash/noop";
+
+//helpers
+import {
+  getCartWithProducts,
+  getTotalAmountOfCart,
+  getTotalItemsInCart,
+} from "./cart.helpers";
+
+// components
+import Button, { TYPES } from "../button";
+import CartItems from "./components/cartItems";
+
+// css
+import "./cart.css";
+import { useMemo } from "react";
+
+const Cart = (props) => {
+  const {
+    cart: { items },
+    products,
+    onCartChange,
+    onCheckout,
+  } = props;
+
+  const itemsCount = getTotalItemsInCart(items);
+
+  const cartWithProducts = useMemo(() => {
+    return getCartWithProducts(products, items);
+  }, [products, items]);
```

Comment:
here we confused between products and items because both gives same kind meaning. you could specify as cartItems and food Items/list or something like that